src/planscape/forsys/get_forsys_inputs.py

```diff
@@ -143,3 +146,116 @@ def _check_project_area_from_url_params_fields_exist(
+    FORSYS_COST_HEADER = "cost"
+
+    # Header prefixes for conditions and priorities.
+    _CONDITION_PREFIX = "c_"
+    _PRIORITY_PREFIX = "p_"
+
+    # List of headers for priorities.
+    # Downstream, this must be in the same order as constructor input
+    # priorities.
+    priority_headers: list[str]
+
+    def __init__(self, priorities: list[str]) -> None:
+        self.priority_headers = []
+
+        for p in priorities:
+            self.priority_headers.append(self.priority_header(p))
+
+    # Returns a priority header givn a priority string.
+    def priority_header(self, priority: str) -> str:
+        return self._PRIORITY_PREFIX + priority
+
+    # Reteurns a condition hader given a condition string.
+    def condition_header(self, condition: str) -> str:
+        return self._CONDITION_PREFIX + condition
+
+
+class ForsysProjectAreaRankingInput():
+    # A dictionary representing a forsys input dataframe.
+    # In the dataframe, headers correspond to ForsysInputHeaders headers. Each
+    # row represents a unique stand.
+    # Dictionary keys are dataframe headers. Dictionary values are lists
+    # corresponding to columns below each dataframe header.
+    forsys_input: dict[str, list]
+
+    def __init__(
+            self, params: ForsysProjectAreaRankingRequestParams,
+            headers: ForsysInputHeaders) -> None:
+        region = params.region
+        priorities = params.priorities
+        project_areas = params.project_areas
+
+        base_condition_ids_to_names = self._get_base_condition_ids_to_names(
+            region, priorities)
+        conditions = self._get_conditions(base_condition_ids_to_names.keys())
+
+        self.forsys_input = self._get_initialized_forsys_input(
+            headers, priorities)
+
+        for proj_id in project_areas.keys():
+            geo = get_raster_geo(project_areas[proj_id])
+
+            self.forsys_input[headers.FORSYS_PROJECT_ID_HEADER].append(proj_id)
+            self.forsys_input[headers.FORSYS_STAND_ID_HEADER].append(proj_id)
+            self.forsys_input[headers.FORSYS_AREA_HEADER].append(geo.area)
+            # TODO: figure out the right value for the units: 5000 is just a
+            # placeholder.
+            self.forsys_input[headers.FORSYS_COST_HEADER].append(
+                geo.area * 5000)
+            for c in conditions:
+                # TODO: replace this with select_related.
+                name = base_condition_ids_to_names[c.condition_dataset_id]
+                score = compute_condition_score_from_raster(
+                    geo, c.raster_name)
+                if score is None:
+                    raise Exception(
+                        "no score was retrieved for condition, %s" % name)
+                self.forsys_input[headers.condition_header(name)].append(score)
+                # TODO: fix this to be sum(1.0 - score) rather than
+                # mean(1.0 - score)
+                self.forsys_input[headers.priority_header(
```

Comment:
why do we include both values here? can we add a comment about it?

Reply:
Actually: the condition score wasn't necessary. I'll remove it.

One of the forsys parameters allows us to filter individual stands by condition scores - including condition scores would only be useful if we want to use those parameters.